### PR TITLE
[Az.Batch] March update

### PR DESCRIPTION
**Pull request by wiboris**
## Mandatory Checklist

- Please choose the target release of Azure PowerShell
  - [x] [General release](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Public preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Private preview](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] [Engineering build](../blob/main/CONTRIBUTING.md#target-release-types)
  - [ ] N/A

- [x] Check this box to confirm: **I have read the [_Submitting Changes_](../blob/main/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/main/CONTRIBUTING.md) and reviewed the following information:**

* **SHOULD** update `ChangeLog.md` file(s) appropriately
    * For SDK-based development mode, update `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`.
        * A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header in the past tense. 
    * For autorest-based development mode, include the changelog in the PR description.
    * Should **not** change `ChangeLog.md` if no new release is required, such as fixing test case only.
* **SHOULD** regenerate markdown help files if there is cmdlet API change. [Instruction](../blob/main/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)
* **SHOULD** have proper test coverage for changes in pull request.
* **SHOULD NOT** adjust version of module manually in pull request

### CODE REVIEW

File: tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv [hunk 1]
```diff
@@ -0,0 +1,142 @@
+Target,Description
+Get-AzApiManagementProperty.md,No synopsis found
+Get-AzApiManagementProperty.md,No description found
+Remove-AzApiManagementAuthorizationServer.md,No examples found
+Update-AzApiManagementRegion.md,No examples found
+Remove-AzAutomationDscConfiguration.md,No examples found
+Remove-AzAutomationDscNodeConfiguration.md,No examples found
```
Comment (wyunchi-ms) [inline tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv:7]: Any reason to add this file? Please remove it.

File: tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv [hunk 2]
```diff
@@ -0,0 +1,142 @@
+Target,Description
```
Comment (vidai-msft) [inline tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv:1]: May I know why you added this file?

File: tools/BatchModelGenerator/Program.cs [hunk 3]
```diff
@@ -392,13 +399,20 @@ private static void GenerateGenericCollectionProperty(PropertyInfo property, str
...
                 // Fill the list by individually wrapping each item in the loop
-                if (magicalDictConversion)
+                if (magicalDictConversion && !regularDictoary)
                 {
                     var keyReference = new CodePropertyReferenceExpression(enumeratorCurrent, OMToPSDictionaryConversionMappings[argType.FullName].Item1);
                     var valueReference = new CodePropertyReferenceExpression(enumeratorCurrent, OMToPSDictionaryConversionMappings[argType.FullName].Item2);
                     CodeMethodInvokeExpression addToList = new(reference, "Add", keyReference, valueReference);
                     loopStatement.Statements.Add(addToList);
                 }
+                else if (regularDictoary) // special case to handel pure Dictionary<string,string> type
```
Comment (wyunchi-ms) [inline tools/BatchModelGenerator/Program.cs:409]: What's this file for?

File: tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv (hunk 2 above)
Comment (wiboris) [reply tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv]: This was a file that got generated when I was trying to build the help files via one of our tools.  I removed it.

File: tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv (hunk 1 above)
Comment (wiboris) [reply tools/StaticAnalysis/Exceptions/ValidateHelpIssues.csv]: removed

File: tools/BatchModelGenerator/Program.cs (hunk 3 above)
Comment (wiboris) [reply tools/BatchModelGenerator/Program.cs:409]: Batch has custom code generation on top of the default powershell generation which needed an update to handle the newest batch C# dataplane sdk.   In the most recent batch C# dataplane sdk we introduced a property of type Dictionary<string,string> in which we have never had such a type before.

**Review by vidai-msft — APPROVED** [review]
LGTM

**Pull request merged**